packages/teleterm/src/ui/services/connectionTracker/connectionTrackerService.test.ts

```diff
@@ -78,3 +85,75 @@ it('removeItemsBelongingToRootCluster removes connections', () => {
     { clusterName: 'remote_leaf', ...connections[3] },
   ]);
 });
+
+it('updates the port of a gateway connection when the underlying doc gets updated', () => {
```

Comment:
@avatus I'm curious what you think about the way we write tests. I've never really TypeScript tests in the past so the whole `X as jest.MockedClass<typeof X>` is new to me and I'm not sure if there's a better way.

Arguably `ConnectionTrackerService` is quite complex, as shown by how complex the test setup is.

Reply:
I see this normally when using typescript to make writing the tests easier (the mocks now have the right class signatures). This is a complex test but I don't really know of a better way (also not the strongest with deep deep testing)

Reply:
Casting here is not needed. StatePersistenceService and ClustersService can be used directly

Reply:
But then the tests would use the real implementation of `ClustersService` and `StatePersistenceService`, wouldn't they?

I do want to mock those objects. They are not the subjects of the test. For the most part I don't care how they behave as long as they fulfill the contract they have with `ConnectionTrackerService`, that is `StatePersistenceService.prototype.getConnectionTrackerState` returns an object of a particular shape.

Reply:
> But then the tests would use the real implementation of ClustersService and StatePersistenceService, wouldn't they?

No. `as` keyword in typescript changes the only type of expression.

Reply:
To mock them you would need 
```
jest.mock('../statePersistence', () => ({
    StatePersistenceService: ... // you mock implementation
}))
```

Reply:
Ah, sorry, I misunderstood you completely.

So I just call `jest.mock(path)` and then use automocked classes from that path just as if `jest.mock()` wasn't called at all?

I checked the docs and it looks like `jest.MockedClass` was removed from the docs starting with Jest v27.

Reply:
I only mean that in
```ts
const StatePersistenceServiceMock =
    StatePersistenceService as jest.MockedClass<typeof StatePersistenceService>;
  const ClustersServiceMock = ClustersService as jest.MockedClass<
    typeof ClustersService
  >;
```
we don't need `ClustersService` and `StatePersistenceService` to have `jest.MockedClass<...>` type because we don't use special `jest.MockedClass` methods/properties of that type